packages/xchain-ethereum/src/client.ts

```diff
@@ -344,11 +344,13 @@ export default class Client implements XChainClient, EthereumClient {
       if (account.tokens) {
         account.tokens.forEach((token) => {
           const decimals = parseInt(token.tokenInfo.decimals)
-          balances.push({
-            asset:
-              assetFromString(`${AssetETH.chain}.${token.tokenInfo.symbol}-${token.tokenInfo.address}`) || AssetETH,
-            amount: baseAmount(token.balance, decimals),
-          })
+          const tokenAsset = assetFromString(`${AssetETH.chain}.${token.tokenInfo.symbol}-${token.tokenInfo.address}`)
```

Comment:
It's still unsafe. Add a validation to check symbol is not undefined or an empty string with a min. length of 3. For validating token address use `validateAddress`.

Reply:
`AssetETH.chain` -> `ETHChain` ;)